src/core/etl/src/Flow/ETL/Rows.php

```diff
@@ -391,7 +383,7 @@ public function joinInner(self $right, Expression $expression) : self
     /**
      * @throws InvalidArgumentException
      */
-    public function joinLeft(self $right, Expression $expression) : self
+    public function joinLeft(self $right, Expression $expression, EntryFactory $entryFactory = new EntryFactory()) : self
```

Comment:
I think we need to stop creating new EntryFactory and simply make it mandatory to not miss any place, as this would be impossible to debug later :|